include/knowhere/version.h

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#pragma once
+
+#include <strings.h>
+
+#include <regex>
+
+#include "log.h"
+
+namespace knowhere {
+namespace {
+static constexpr const char* version_regex = R"(^(knowhere)-[vV]\d+\.\d+\.\d+$)";
+static constexpr const char* default_version = "knowhere-v2.2.0";
+static constexpr const char* minimal_vesion = "knowhere-v0.1.0";
```

Comment:
suggest to refer to faiss:
#define FAISS_VERSION_MAJOR 1
#define FAISS_VERSION_MINOR 7
#define FAISS_VERSION_PATCH 2